### PR TITLE
Checkout: update support link

### LINKED ISSUE #49150: Link to initiate Support Chat is overlapped by upsell box in checkout
### Steps to reproduce the behavior

Prereq: Use a free site with no purchases.
1. Go to My Site.
2. Add domain to site.
3. Choose a custom domain.
4. Add to cart and check out.
5. Skip G Suite integration.
6. At secure checkout, inspect the right pane.

#### What I expected to happen
Link to initiate Support Chat should be clearly visible.

#### What actually happened
`Upgrade and Save` upsell box overlaps half of the link.

### Context
Manual user testing.

#### Browser / OS version
Browser: Firefox Nightly 86.0a1
OS: macOS Catalina 10.15.7

#### Is this specific to the applied theme? Which one?
N/A.

#### Does this happen on simple or atomic sites or both?
N/A.

#### Is there any console output or error text?
No.

#### Level of impact (Does it block purchases? Does it affect more than just one site?)
Minor - user is still able to initiate chats through the `?` icon on the bottom right if needed.

#### Screenshot / Video: If applicable, add screenshots to help explain your problem.
Screen Shot 2021-01-21 at 12 03 20

### PR DESCRIPTION
**Pull request by michaeldcain**
In #48695, we fixed the right margin of the Checkout chat link when inside the `CheckoutSummary`, but when the monthly plans rollout happened, the link was absolutely positioned below the summary, creating a number of display issues. This PR bundles the `CheckoutSupportLink` component (logic and styles) with the `PaymentsChatButton`, renames that files, and properly includes it one level up (inside of `CheckoutSummaryBody`), beneath the upsells (when visible).

Fixes: #49150

**Chat eligible:**
Screen Shot 2021-01-27 at 11 51 31 AM

**Chat eligible (with upsell):**
Screen Shot 2021-01-27 at 11 50 58 AM

**Chat eligible on mobile (with upsell):**
Screen Shot 2021-01-27 at 11 51 10 AM

**Chat not eligible:**
Screen Shot 2021-01-27 at 11 56 35 AM

**To test:**
- force the pre-sales HappyChat button to render by setting [this](https://github.com/Automattic/wp-calypso/blob/fix/checkout-chat-button/client/components/happychat/button.jsx#L83) and [this](https://github.com/Automattic/wp-calypso/blob/fix/checkout-chat-button/client/my-sites/checkout/composite-checkout/components/checkout-help-link.tsx#L153) to true
- add a domain to your cart to visit Checkout
- verify that the button matches the screenshots above (with the upsell)
- click the upsell to add a plan to your cart
- verify that the button matches the screenshots above (without the upsell)

### CODE REVIEW

File: client/my-sites/checkout/composite-checkout/components/checkout-help-link.tsx [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+				plan,
+				support_level: supportLevel,
+			} )
+		);
+	};
+
+	return (
+		<HappychatButton onClick={ chatButtonClicked }>
+			<Gridicon icon="chat" />
+			{ translate( 'Need help? Chat with us.' ) }
```
Comment (a8ci18n) [inline client/my-sites/checkout/composite-checkout/components/checkout-help-link.tsx:73]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **29** times:
`translate( 'Need help? Chat with us' )` *ES Score: 10*

File: client/my-sites/checkout/composite-checkout/components/checkout-help-link.tsx [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+} from 'calypso/state/selectors/get-inline-help-support-variation';
+import {
+	isWpComBusinessPlan,
+	isWpComEcommercePlan,
+	isWpComPersonalPlan,
+	isWpComPremiumPlan,
+} from 'calypso/lib/plans';
+
+type StyledProps = {
+	theme?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/components/checkout-help-link.tsx]: I think we can give this a type actually by importing `Theme` from `@automattic/composite-checkout`. I'll give it a try.

**Review by sirbrillig — APPROVED** [review]
Works as described!

File: client/my-sites/checkout/composite-checkout/components/checkout-help-link.tsx (hunk 2 above)
Comment (sirbrillig) [reply client/my-sites/checkout/composite-checkout/components/checkout-help-link.tsx]: I added the type.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5477265).

**Pull request merged**